### PR TITLE
Upgrade the project to Swift 4.2 and Xcode 10.1

### PR DESCRIPTION
**Pull request by Vkt0r**
Hey guys 👋,

Swift 5 is coming soon so I think is a good option to keep the project updated to Swift 4.2 and Xcode 10.1. This PR can be resumed in the following:

* Upgrade `RxSwift`, `RxCocoa` to 4.4.1.
* Upgrade `Crashlytics` to 3.12.0.
* Upgrade `Device` to  3.1.2.

### CODE REVIEW

**Comment by freak4pc** [comment]
Pinging @yury here as well as he's sort of the lead person on this app

File: RxMarbles/SceneView.swift [hunk 1]
```diff
@@ -122,7 +122,7 @@ class SceneView: UIView, UITextViewDelegate {
         
         let size = rxOperatorText.text.boundingRect(with: CGSize(width: bounds.width, height: 1000 ),
                                                     options: NSStringDrawingOptions.usesLineFragmentOrigin,
-                                                    attributes: [NSAttributedStringKey.font: rxOperatorText.font!],
+                                                    attributes: [NSAttributedString.Key.font: rxOperatorText.font!],
```
Comment (freak4pc) [inline RxMarbles/SceneView.swift]: can we just do
```suggestion
                                                    attributes: [.font: rxOperatorText.font!],
```

File: RxMarbles/OperatorViewController.swift [hunk 2]
```diff
@@ -162,8 +162,8 @@ class OperatorViewController: UIViewController, UISplitViewControllerDelegate {
         let activity = UIActivityViewController(activityItems: [_makeSnapshot()], applicationActivities: nil)
         
         activity.excludedActivityTypes = [
-            UIActivityType.assignToContact,
-            UIActivityType.print,
+            UIActivity.ActivityType.assignToContact,
+            UIActivity.ActivityType.print,
```
Comment (freak4pc) [inline RxMarbles/OperatorViewController.swift]: can we do :
```suggestion
            .print,
```

File: RxMarbles/OperatorImplementation.swift [hunk 3]
```diff
@@ -791,10 +791,10 @@ extension Operator {
     
     var linkText: NSMutableAttributedString {
         let res = NSMutableAttributedString(string: text)
-        let link = NSMutableAttributedString(string: "Read\u{a0}more...", attributes: [NSAttributedStringKey.link: url])
+        let link = NSMutableAttributedString(string: "Read\u{a0}more...", attributes: [NSAttributedString.Key.link: url])
```
Comment (freak4pc) [inline RxMarbles/OperatorImplementation.swift]:
```suggestion
        let link = NSMutableAttributedString(string: "Read\u{a0}more...", attributes: [.link: url])
```

File: RxMarbles/OperatorImplementation.swift [hunk 4]
```diff
@@ -791,10 +791,10 @@ extension Operator {
...
         res.append(NSAttributedString(string: " "))
         res.append(link)
-        res.addAttribute(NSAttributedStringKey.font, value: Font.code(.monoRegular, size: 16), range: NSMakeRange(0, res.length))
+        res.addAttribute(NSAttributedString.Key.font, value: Font.code(.monoRegular, size: 16), range: NSMakeRange(0, res.length))
```
Comment (freak4pc) [inline RxMarbles/OperatorImplementation.swift]:
```suggestion
        res.addAttribute(.font, value: Font.code(.monoRegular, size: 16), range: NSMakeRange(0, res.length))
```

File: RxMarbles/OperatorCode.swift [hunk 5]
```diff
@@ -135,30 +135,30 @@ extension Operator {
     func higlightedCode() -> NSAttributedString {
         let font = Font.code(.monoRegular, size:18)
         let src = NSMutableAttributedString(string: code, attributes: [
-            NSAttributedStringKey.font: font,
-            NSAttributedStringKey.foregroundColor: Color.codeDefault
+            NSAttributedString.Key.font: font,
```
Comment (freak4pc) [inline RxMarbles/OperatorCode.swift]:
```suggestion
            .font: font,
```

and everywhere else in this file if you can

**Review by freak4pc — COMMENTED** [review]
Looks good ! Left a bit of comments.
Let's try getting rid of the verbosiy of `NSAttributedString.Key.something` for `.something` - if that was the case before there would barely be any migration needed :)

**Comment by yury** [comment]
Cool! Thank you @Vkt0r.

I will look at this PR a little bit later today.

**Comment by Vkt0r** [comment]
Cool @yury! Glad to help. @freak4pc I committed your suggestions and squashed to keep it simple.

**Comment by yury** [comment]
@Vkt0r,  awesome work. Thanks

@freak4pc thanks for heads up!

Will try to release soon!

**Pull request merged**